### PR TITLE
Item#canEntityPickup

### PR DESCRIPTION
**Pull request by BillyGalbreath**
This is related to #679 due to the fact that cancelling the EntityPickupItemEvent can be somewhat spammy (~15 calls per second). I decided to put the two in different PRs because they both have separate use-cases where they do not rely on each other (event for listening, flag for blocking).

With this PR you can set an Item to not be picked up by a non-player entity, such as a money drop plugin that uses gold nuggets/ingots/blocks as dropped currency, for example. This flag is beneficial over #679 due to the fact is skips a lot of logic and does not call a spammy event which may have a listener with its own heavy logic. The flag can be set at the time the Item was created and then forgotten about.

### CODE REVIEW

**Comment by Black-Hole** [comment]
There exists a special value for pickup delay for this purpose already: Short.MAX_VALUE (32767)
It's a vanilla feature and undocumented in the Item class.

**Comment by BillyGalbreath** [comment]
The EntityItem#pickupDelay int isn't the same thing as this. Setting the pickupDelay will prevent entities _and_ players from picking up the item. This PR only effects non-player entities.

**Comment by zachbr** [comment]
I'm a little concerned about confusion related to the non-player specificity of this API, but open to it.

**Review by kashike — APPROVED** [review]

File: Spigot-API-Patches/0051-Item-canEntityPickup.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      */
+     public void setPickupDelay(int delay);
++
++    // Paper Start
++    /**
++     * Gets if non-player entities can pick this Item up
++     *
++     * @return True if non-player entities can pickup
++     */
++     public boolean canEntityPickup();
```
Comment (kashike) [inline Spigot-API-Patches/0051-Item-canEntityPickup.patch]: `canMobPickup` might be a better name

**Review by zachbr — APPROVED** [review]

**Pull request merged**